Add tooltips to popup buttons

Gigahawk opened the pull request:
Adds tooltups when long pressing popup buttons:
![image](https://user-images.githubusercontent.com/10356230/212822626-f6c3ab21-4715-4b13-8114-0984e80a5f4b.png)

mrkuz committed:
Add preference to hide empty days in agenda.

Resolves #516

dwoffinden committed:
Android R+: request MANAGE_EXTERNAL_STORAGE when creating new git repos

amberin committed:
Git via SSH: Replace Jsch transport with Apache

N.B. Requires API >= 26.

Apache MINA SSHD is now the standard SSH transport library for JGit. It
supports more modern key algorithms than Jsch.

I have added a notification prompt to the user upon new or unexpected SSH
server host key. I have no idea what I'm doing, but it seems to work.

amberin committed:
Remove unused methods

amberin committed:
Generate the SSH key for Git syncing on the device

A generated key can optionally be protected by biometric auth or device
credential. This makes it harder to steal, but will obviously not play
well with auto-sync.

The default key type is EDCSA. ED25519 keys are faster, but not
supported natively by the Android key store. The methods currently
called when unlocking a ED25519 key do not respect the validity duration
setting, which means that the key needs to be unlocked before each use.
This may be twice during a sync, if we need to both fetch and push. RSA
and EDCSA keys respect the validity duration setting, meaning we leave
them unlocked for 15 seconds.

A way to speed up Git syncing while requiring auth upon each key use
would be to use SSH multiplexing and keep the SSH session open until we
push (or decide not to push).

I raised the minimum SDK version from 21 to 23. Otherwise we cannot
include android-crypto in the manifest.

N.B. Much of this code has been taken and re-worked from
https://github.com/android-password-store/Android-Password-Store. That
project is also GPL-3.0, but I don't know how to properly attribute
those authors in the Orgzly code base.

amberin committed:
Upgrade JGit to 5.13.1

amberin committed:
Improve GitFileSynchronizer

- Make "git push" run blocking. Mostly to make the "syncing in progress"
status information more reliable.
- Show more informative snackbar when "git push" goes wrong. The user
will now find out if pushing fails because of e.g. missing write
permission.
- Remove two unused methods and an unused return value.
- Tidy up some variables.

amberin committed:
Improve error handling during cloning

The TransportException class path was wrong, so the exception type was
not being recognized.

Make an attempt to find the cause if JGit throws the generic "remote
hung up unexpectedly" exception.

amberin committed:
Allow Git repos to be updated

We were always ending up in RepoCloneTask.

Separate two different error cases, so that we can actually distinguish
between them.

amberin committed:
Populate the preference fields with some sane defaults

Also, use more specific inputTypes.

Also, fix the branch name always showing as the default value

amberin committed:
Git sync: Push current branch to remote if it is empty

This allows the user to easily set up a new "origin" remote (by
initalizing an empty repository) if the current one is somehow lost.

A Git repo which is being synced between two computers should provide a
degree of redundancy. Without this code, there was no simple way to
sync the repo contents using Orgzly as the source.

Gigahawk committed:
Add tooltips to popup buttons

Gigahawk commented on app/src/main/java/com/orgzly/android/prefs/NotePopupPreference.kt:
```diff
@@ -45,10 +45,10 @@ class NotePopupPreference : DialogPreference {
         private val dividerLeft = NotePopup.Action(
-            R.id.note_popup_divider, R.drawable.ic_swipe_left, "divider")
+            R.id.note_popup_divider, R.drawable.ic_swipe_left, R.string.ok, "divider")
 
         private val dividerRight = NotePopup.Action(
-            R.id.note_popup_divider, R.drawable.ic_swipe_right, "divider")
+            R.id.note_popup_divider, R.drawable.ic_swipe_right, R.string.ok, "divider")
```
Not really sure where these buttons show up or what an appropriate tooltip would be for these buttons

amberin committed:
Simplify GitRepo.retrieveBook()

The method had several unused variables and seemingly pointless logic,
considering how it is used.

The method is only called in the current scenarios:

- A remote book has no corresponding local book
- When force loading a book from the repo

Hence, I see no reason to "safely" retrieve the latest version of the
book from the repo. We know that we just want the latest version from
the repo.

amberin committed:
Argument has been renamed in JGit

amberin committed:
Simplify getLatestCommitOfFile

This method seems to be very expensive, we probably need to optimize
this stuff.

amberin committed:
Clean up two instances of unused exceptions

amberin committed:
Add missing call() when deleting temporary branches

This may explain why the local repo kept growing quickly and temporary
branches were not being deleted.

Also, add some more exception handling, debug logging and elaborate
error messages.

amberin committed:
General cleanup, adding some debug logging

amberin committed:
Git: Implement "delete" method from the SyncRepo interface

amberin committed:
Git: Implement renameBook method from the SyncRepo interface

amberin committed:
Git: Simplify sync when there are only local changes

We should be able to trust the namesake status. If there are only local
changes, there is no need for complex two-way syncing. The
handleTwoWaySync method call resulted in the local notebook always being
re-loaded into the database from the Git repo, even though we just wrote
that version ourselves. This was a frustrating user experience, since
each sync of local changes resulted in all notes objects being replaced
in the database, which made it impossible to initiate sync and keep on
adding notes in the same book.

If there are only local changes, we now use the
dataRepository::saveBookToRepo method, just like other repo types.

The only drawback of this change is that we have to "git push" in the
GitRepo::storeBook method, since it is also used during force save of a
book. This means that the GitRepo::tryPushIfHeadDiffersFromRemote is
later run in vain, as it is run after each syncing of a Git repo. No
push will be performed, though, and I have now removed the pointless
looping over remote branches which I once added.

amberin committed:
Skip Git repo walk if the current head has not moved

If nothing has changed in the Git repo, we can save a lot of sync time
by reading vrook information from the database.

amberin committed:
Handle changed user device name on API >= 32

chvp committed:
Show ranged timestamps in agenda queries

chvp committed:
Consider time and date ranges when querying events

colonelpanic8 committed:
Merge remote-tracking branch 'dwoffinden/manager' into imalison-get-git-working

amberin committed:
Change the app name and ID

I deleted the Play store and F-Droid buttons, since we don't currently
publish to either.

cbkerr committed:
Add sync icon to widget

I merged in updates to the PR by @tomoyukim.

cbkerr committed:
Merge branch 'master' into widget-sync

cbkerr committed:
Clean up line ordering changes

cbkerr committed:
Two more order cleanups

colonelpanic8 committed:
Merge pull request #12 from cbkerr/widget-sync

Add sync icon to list widget

amberin committed:
Add workflows to build, sign and release APK

The unusual thing here is the workflow for creating JKS keystore as a
repo secret.

The idea is to use Github Actions to generate the key pair used for
signing the APKs, and store the resulting key store as secrets in the
repository, available for use by the "build and release" actions.

This way, no one contributor needs to be entrusted with key generation,
minimizing the opportunities for any one person to keep their own copy
of the app signing key.

The only way to read repository secrets is to create and run an
Action/workflow where the secrets are printed in an roundabout manner
(because Github tries hard to keep ensure that they are always masked).
This can of course be done by repo/organization admins, but it is hard
to do without leaving any traces.

In order to be allowed to create repository secrets from a workflow, a
fine-grained personal access token is required when running the "Create
APK signing secrets" workflow (here given the name
SECRETS_CREATION_TOKEN). Since this workflow is only really intended to
be run once, and then disabled (but kept as documentation), I recommend
creating a very short-lived personal access token for use in this
workflow. It should not matter who it belongs to (but it obviously needs
to be a repo/org admin).

If anyone else wants to trigger the workflow to overwrite the app
signing keys, they must have repo/org admin privileges, so that they can
first create a personal access token with permission to write secrets.

amberin committed:
Add monochrome icon definition

Since the regular icon foreground is monochrome, I just pointed to that,
and it seems to work fine.

amberin committed:
Don't generate APK signing secrets from Github Action

I have concluded that this was a bad idea due to the high risk of losing
the secret material.

We will instead create these secrets locally and share them between the
maintainers, so that there are multiple copies. We will then add them as
repo secrets, so the end result will be the same.

Having access to the secret material will also make it easier for us to
rotate the signing key, if/when necessary.

Please see the discussion in PR #16 for more details.

amberin committed:
Polish release and prerelease build workflows

- Better names for the uploaded APK files
- Always create "draft" Github releases (to avoid accidental/premature publishing)
- Run zipalign on the release build
- Use the "premium" (Google Play) flavor for both builds

Publishing "pre-release" builds is a new practice for Orgzly. Let's
evaluate its usefulness. I'm thinking that end users may appreciate the
possibility to beta test upcoming releases. This is why I have chosen to
stick with the same flavor as in the release artifacts -- to allow
seamless updates between beta and GA versions. This will give us more
eyeballs on upcoming releases.

The release artifacts published on Github have always been of the
"premium" (Google Play) flavor so far, so let's stick to that.

colonelpanic8 committed:
Merge pull request #16 from amberin/apk-signing-and-release-workflows

Add workflows to build, sign and release APK

amberin committed:
Merge pull request #18 from amberin/monochrome-icon

Add monochrome icon definition

amberin committed:
Merge pull request #19 from amberin/appname

Change the app name and ID

amberin committed:
Update README with new organization and repo name

amberin committed:
Update version to 173 (1.8.11-beta.2)

Version 172 was created in a branch which was never merged to master,
but a release was created -- hence why I'm skipping it now.

amberin committed:
Add changes for v1.8.11-beta.2

amberin committed:
Keystore entry alias is not named "mykey"

I named this entry "orgzly-revived-20231013", in case we ever need to
rotate the key. We can then keep multiple key entries in the same JKS,
if we wish. In any case, the name now tells when the key was created.

I also just noticed that the action r0adkll/sign-android-release@v1 also
performs zipalign, so we might as well use that everywhere.

amberin committed:
Workflows need write permission to create a release

amberin committed:
Simplify the names of the new workflows

amberin committed:
Don't run android-build-master upon push to branch release-*

Because we now have dedicated (pre)release workflows.

amberin committed:
Git: Ensure repo is explored for notebooks if no linked books are found

This solves the most pressing problem in #22, which was that no notebooks were
loaded after cloning a repo (since no new commits were fetched).

amberin committed:
Merge pull request #26 from amberin/issue-22-first-sync

Git: Ensure repo is explored for notebooks if no linked books are found

amberin committed:
Merge branch 'master' into release-1.8.11

amberin committed:
Update version to 174 (1.8.11-beta.3)

cbkerr commented:
Hi @Gigahawk, are you interested in applying this onto [Orgzly Revived](https://github.com/orgzly-revived/orgzly-android-revived)? It's a community fork just getting off the ground.

amberin committed:
Merge pull request #32 from chvp/fix/agenda-ranged-timestamps

Show ranged timestamps in agenda queries

amberin committed:
Merge pull request #33 from chvp/feat/event-range-queries

Consider time and date ranges when querying events

cbkerr committed:
Merge remote-tracking branch 'mrkuz/feature/hide-empty-days-in-agenda' into agenda-hide-empty

amberin committed:
Update version to 175 (1.8.11)

amberin committed:
Merge pull request #36 from orgzly-revived/release-1.8.11

Release 1.8.11

jojhelfer committed:
Add menu option to sort lines in a note

alensiljak committed:
#7 gradle sha256

amberin committed:
Merge pull request #41 from alensiljak/f-droid

#7 gradle sha256

cbkerr committed:
Update whats new intro and keep link to original repo

colonelpanic8 committed:
Merge pull request #40 from cbkerr/agenda-hide-empty

Hide empty days in agenda

colonelpanic8 committed:
Merge pull request #43 from cbkerr/update-changelog-git-link

Update whats new intro and keep link to original repo

alensiljak committed:
expanding on the project description

alensiljak committed:
Merge branch 'orgzly-revived:master' into master

alensiljak committed:
link in org format

Co-authored-by: Corwin Kerr <corwin.kerr@gmail.com>

alensiljak committed:
repo link in Org format

Co-authored-by: Corwin Kerr <corwin.kerr@gmail.com>

alensiljak committed:
orgzly link in Org format

Co-authored-by: Corwin Kerr <corwin.kerr@gmail.com>

alensiljak committed:
"Contributing" section

alensiljak committed:
purpose

alensiljak committed:
link to repo in the description

cbkerr committed:
Grammar fix

amberin committed:
Merge pull request #44 from alensiljak/master

Project description

amberin committed:
Update issue templates

- The file .github/ISSUE_TEMPLATE.md is deprecated.
- Let's stick to the default issue labels for now.

amberin committed:
Merge pull request #48 from amberin/issue-templates

Update issue templates

amberin committed:
Merge pull request #37 from jojhelfer/master

Add menu option to sort lines in a note

Riyyi committed:
Add auto-sync on suspend

Trigger auto-sync when the app becomes invisible to the user,
ex: when navigating away or when locking the device.

amberin committed:
Merge pull request #50 from Riyyi/master

Add auto-sync on suspend

Gigahawk commented:
> Hi @Gigahawk, are you interested in applying this onto [Orgzly Revived](https://github.com/orgzly-revived/orgzly-android-revived)? It's a community fork just getting off the ground.

Migrated to https://github.com/orgzly-revived/orgzly-android-revived/pull/52
